src/Microsoft.DotNet.Build.Tasks/VersionTools/UpdateDependenciesAndSubmitPullRequest.cs

```diff
@@ -121,5 +126,23 @@ private ITaskItem FindDependencyBuildInfo(string name)
         {
             return DependencyBuildInfo.SingleOrDefault(item => item.ItemSpec == name);
         }
+
+        private void ReplaceProperty(string path, string elementName, string newValue)
+        {
+            const string valueGroup = "valueGroup";
+            Action replaceAction = FileUtils.ReplaceFileContents(
+                path,
+                contents =>
+                {
+                    Group g = CreateXmlUpdateRegex(elementName, valueGroup)
+                        .Match(contents)
+                        .Groups[valueGroup];
+
+                    return contents
+                        .Remove(g.Index, g.Length)
+                        .Insert(g.Index, newValue);
+                });
+            replaceAction?.Invoke();
```

Comment:
`ReplaceFileContents` only returns an `Action` if there's a change to make, and there might not be a change. For example: if `AutoUpgradeBranch` is `master` but the `CurrentBranch` is already `master`.

@naamunds suggested recently that I rename the `ReplaceFileContents` method, and I think that would help here. `GetReplaceFileContentsTask` is what I'm thinking, paralleling `IDependencyUpdater::GetUpdateTasks`. I'll add a comment too because it's not clear here if not making a replacement is actually an upgrade error.